tools/export/uvision4.py

```diff
@@ -73,10 +73,14 @@ def generate(self, progen_build=False):
 
         # get flags from toolchain and apply
         project_data['tool_specific']['uvision']['misc'] = {}
-        # asm flags only, common are not valid within uvision project, they are armcc specific
-        project_data['tool_specific']['uvision']['misc']['asm_flags'] = list(set(self.progen_flags['asm_flags']))
+        # need to make this a string for progen. Only adds preprocessor when "macros" set
+        asm_flag_string = '--cpreproc --cpreproc_opts=-D__ASSERT_MSG,' + ",".join(
```

Comment:
Progen does this if [common macros](https://github.com/project-generator/project_generator/blob/master/project_generator/tools/uvision.py#L251) are defined, to be used with armcc and armasm. Because I am trying to distinguish between the two (without editing progen), I wanted to format this entire line to be used in [this progen line](https://github.com/project-generator/project_generator/blob/master/project_generator/tools/uvision.py#L261).